src/dmd/cppmangle.d

```diff
@@ -1296,10 +1296,14 @@ extern(C++):
                 // Handle any target-specific basic types.
                 if (auto tm = Target.cppTypeMangle(t))
                 {
-                    if (substitute(t))
-                        return;
-                    else
-                        append(t);
+                    // Only do substitution for mangles that are longer than 1 character.
+                    if (tm[1] != 0 || t.isConst())
```

Comment:
Hmm, I noticed this discrepancy with `writeBasicType` when submitting my ldc pull, traced both changes back to Walter's pull, saw he added both blocks, and figured that's what he wanted, since I didn't know what this substitution was about. I should have asked about it then.